inst/shiny/modules/column_mapping.R

```diff
@@ -1,3 +1,42 @@
+#' Column Mapping Widget
+#'
+#' A reusable UI component for mapping dataset columns to specific identifiers or roles.
+#'
+#' @param ns A namespace function to generate IDs for Shiny inputs.
+#' @param id A string representing the unique identifier for the widget.
+#' @param tooltip_text A string containing the tooltip text to guide users.
+#'
+#' @return A Shiny `div` containing a `selectizeInput` with associated labels and tooltip.
+#'
+#' @examples
+#' column_mapping_widget(ns = NS("example"), id = "STUDYID",
+#' tooltip_text = "Select the study identifier column.")
+#'
+#' @export
+column_mapping_widget <- function(ns, id, tooltip_text) {
+  div(
+    class = "column-mapping-row",
+    tooltip(
+      selectizeInput(
+        ns(paste0("select_", id)),
+        "",
+        choices = NULL,
+        options = list(placeholder = "Select Column"),
+        width = "40%"
+      ),
+      tooltip_text
+    ),
+    div(
+      class = "column-mapping-output",
+      span(paste0(id, ":"), style = "color: black; font-size: normal;")
+    ),
+    div(
+      class = "column-mapping-label",
+      span(textOutput(ns(paste0("label_", id))), style = "color: grey;")
```

Comment:
nitpick: Inline style not needed, styling in CSS file will cover it.
```suggestion
      span(textOutput(ns(paste0("label_", id))))
```